pymc3/tests/test_bart.py

```diff
@@ -0,0 +1,113 @@
+    assert split_node.index == 5
+    assert split_node.idx_split_variable == 2
+    assert split_node.split_value == 3.0
+    assert split_node.depth == 2
+    assert split_node.get_idx_parent_node() == 2
+    assert split_node.get_idx_left_child() == 11
+    assert split_node.get_idx_right_child() == 12
+
+
+def test_leaf_node():
+    leaf_node = pm.distributions.bart.LeafNode(index=5, value=3.14, idx_data_points=[1, 2, 3])
+    assert leaf_node.index == 5
+    assert np.array_equal(leaf_node.idx_data_points, [1, 2, 3])
+    assert leaf_node.value == 3.14
+    assert leaf_node.get_idx_parent_node() == 2
+    assert leaf_node.get_idx_left_child() == 11
+    assert leaf_node.get_idx_right_child() == 12
+
+
+def test_bart():
+    X = np.random.sample(size=(100, 4))
+    Y = np.random.sample(size=(100))
+
+    with pm.Model():
+        b = pm.BART("b", X=X, Y=Y)
+
+    bart = b.distribution
+    assert bart.num_observations == 100
+    assert bart.num_variates == 4
+    assert len(bart.trees) == bart.m
+
+    assert np.array_equal(bart.trees[0][0].idx_data_points, np.array(range(100), dtype="int32"))
+
+
+def test_available_splitting_rules():
+    X = np.array([[1.0, 2.0, 3.0, np.nan], [2.0, 2.0, 3.0, 99.9], [3.0, 4.0, 3.0, -3.3]])
+    Y = np.random.sample(size=(3))
+    with pm.Model():
+        b = pm.BART("b", X=X, Y=Y)
+
+    bart = b.distribution
+
+    idx_split_variable = 0
+    idx_data_points = np.array(range(bart.num_observations), dtype="int32")
+    available_splitting_rules = bart.get_available_splitting_rules(
+        idx_data_points, idx_split_variable
+    )
+    assert available_splitting_rules.size == 2
+    np.testing.assert_almost_equal(available_splitting_rules, np.array([1.0, 2.0]), 1)
+
+    idx_split_variable = 1
+    idx_data_points = np.array(range(bart.num_observations), dtype="int32")
+    available_splitting_rules = bart.get_available_splitting_rules(
+        idx_data_points, idx_split_variable
+    )
+    assert available_splitting_rules.size == 1
+    np.testing.assert_almost_equal(
+        available_splitting_rules,
+        np.array(
+            [
+                2.0,
+            ]
+        ),
+        1,
+    )
+
+    idx_split_variable = 2
+    idx_data_points = np.array(range(bart.num_observations), dtype="int32")
+    available_splitting_rules = bart.get_available_splitting_rules(
+        idx_data_points, idx_split_variable
+    )
+    assert available_splitting_rules.size == 0
+    np.testing.assert_almost_equal(available_splitting_rules, np.array([]))
+
+    idx_split_variable = 3
+    idx_data_points = np.array(range(bart.num_observations), dtype="int32")
+    available_splitting_rules = bart.get_available_splitting_rules(
+        idx_data_points, idx_split_variable
+    )
+    assert available_splitting_rules.size == 1
+    np.testing.assert_almost_equal(available_splitting_rules, np.array([-3.3]), 1)
+
+
+def test_model():
+    np.random.seed(212480)
```

Comment:
Im not sure about this but should random seeds be generated with numpys rng thing?

Reply:
Also not sue, i think this is something we do in many of our tests, so I will defer until we decide on an overall action.